packages/flutter/lib/src/material/dialog.dart

```diff
@@ -808,6 +833,44 @@ class SimpleDialog extends StatelessWidget {
       }
     }
 
+    // The paddingScaleFactor is used to adjust the edge padding of Dialog
```

Comment:
If we're really doing the exact same thing here (?) it might be better to factor out the code.

Reply:
It is very similar. The difference lies in the way paddingScaleFactor is applied to each edge (it is skipped for certain edges). The paddingScaleFactor calculation is factored out into a function, but I can't think of how to further factor this out.